fitbenchmarking/controllers/levmar_controller.py

```diff
@@ -0,0 +1,122 @@
+"""
+Implements a controller for the levmar fitting software.
+http://www.ics.forth.gr/~lourakis/levmar/
+via the python interface
+https://pypi.org/project/levmar/
+"""
+
+import levmar
+
+from fitbenchmarking.controllers.base_controller import Controller
+
+
+class LevmarController(Controller):
+    """
+    Controller for the levmar fitting software
+    """
+
+    def __init__(self, cost_func):
+        """
+        Initialise the class.
+
+        :param cost_func: Cost function object selected from options.
+        :type cost_func: subclass of
+                :class:`~fitbenchmarking.cost_func.base_cost_func.CostFunc`
+        """
+        super(LevmarController, self).__init__(cost_func)
+        self._popt = None
+        self.algorithm_check = {
+            'all': ['levmar', 'levmar-no-jac'],
+            'ls': ['levmar', 'levmar-no-jac'],
+            'deriv_free': [],
+            'general': []}
+        self._info = None
+
+    def jacobian_information(self):
+        """
+        levmar can use Jacobian information
+        """
+        has_jacobian = True
+        jacobian_free_solvers = ["levmar-no-jac"]
+        return has_jacobian, jacobian_free_solvers
+
+    def setup(self):
+        """
+        Setup problem ready to be run with levmar
+        """
+
+        return
+
+    def _feval(self, p, x):
+        """
+        Utility function to call problem.eval_model with correct args
+
+        :param p: parameters
+        :type p: list
+        :param data: x data
+        :type data: list
+        :return: result from problem.eval_model
+        :rtype: numpy array
+        """
+
+        fx = self.problem.function(self.data_x, *p) 
+        return fx
+
+    def _jeval(self, p, x):
+        """
+        Utility function to call jac.eval with correct args
+
+        :param p: parameters
+        :type p: list
+        :param data: x data, this is discarded as the defaults can be used.
+        :type data: N/A
+        :return: result from jac.eval
+        :rtype: numpy array
+        """
+        jac = -self.jacobian.eval(p)
+        return jac
```

Comment:
Pylint throws up an unused argument warning for the x argument in these functions